### PR TITLE
Allow construction of Variables from multi dimensional lists and tuples

### PR DESCRIPTION
**Pull request by jl-wynen**
Also fixes the only outstanding point of https://github.com/scipp/scipp/issues/1541

Example:
 ```.py
var1 = sc.array(dims=['x', 'y'], values=[[1, 2], [3, 4]])
var2 = sc.array(dims=['x', 'y'], values=[(1, 2), (3, 4)])
```

### CODE REVIEW

File: python/make_variable.h [hunk 1]
```diff
@@ -114,18 +114,26 @@ auto do_init_0D(const T &value, const std::optional<T> &variance,
   return var;
 }
 
+// If we use py::array as the type of values and variances, then pybind11 only
+// accepts exactly numpy.array as input and not types that are convertible to
+// an array (e.g. list). But doing the conversion manually using .cast<py:array>
+// allows this function to work with anything that can be converted to an array.
+// The downside is that we make an extra copy if the input is not already an
+// array. But that is likely not important as lists / tuples should not contain
+// large data.
```
Comment (SimonHeybrock) [inline python/make_variable.h:123]: So if it is a *numpy* array no extra copy is made?

File: python/variable.cpp [hunk 2]
```diff
@@ -211,7 +197,6 @@ of variances.)");
         return size_of(self, SizeofTag::Underlying);
       });
 
-  bind_init_list(variable);
   // Order matters for pybind11's overload resolution. Do not change.
   bind_init_0D_numpy_types(variable);
   bind_init_0D_native_python_types<bool>(variable);
```
Comment (SimonHeybrock) [inline python/variable.cpp:202]: Are these still reachable now that the constructor above accepts any `py::object`?

File: python/make_variable.h (hunk 1 above)
Comment (jl-wynen) [reply python/make_variable.h:123]: It might shallow copy the array but it does not copy the underlying buffer.

File: python/variable.cpp (hunk 2 above)
Comment (jl-wynen) [reply python/variable.cpp:202]: Yes. These are for 0d and have a different signature (no `dim` and `value` instead of `values`).

**Review by SimonHeybrock — APPROVED** [review]

**Pull request merged**